### ISSUE TITLE
[Discord] Make option to turn off group chats/channels ; & add manual history sync from IMs/DMs (private chat)

### ISSUE THREAD

**Issue by tealkami** [miranda-ng/miranda-ng#1863]
**Is your feature request related to a problem? Please describe.**
TLDR: name in the Title
I am on 34 "Discord server" most of them have 30-60 channels and few have 100-400 channels
The problem is that I have now 3000 group chats only from Discord account (people tend to create and delete channels), but i just want Sync history for my private chats/single person contacts , i don't need to write/read in group chat in Miranda NG , I can check channel history in Discord program/website/app
Miranda NG have best history search , that's why i like to sync history on Miranda NG

and second request : sometimes it doesn't sync properly , or Miranda NG crash because of many groups ... so like on SkypeWeb have "Get server history" in contact menu

**Describe the solution you'd like**
for groups make something like :
Option=>Netowrk=>Discord account =>  checkmark for "automaticly join group chat on login"
so i can uncheck it
or "don't sync channels at all"

for manual sync make it same as in SkypeWeb "Get server history" , click option for contact menu

**Additional context**
*Add any other context or screenshots about the feature request here.*
why not https://i.imgur.com/obINIbL.png

**Comment by georgehazan** [comment]
yes, guild management should be more effective, for example, web client allows you to use one guild only, all another ones are just listed. maybe we can use the same principle in Miranda?
@LittleVulpix what do you think about it?

**Comment by LittleVulpix** [comment]
It makes sense for sure; it should be configurable. And about use vs list, well, when someone mentions you (or for example @everyone) , the web client alerts you. The web client also shows if there has been a new message posted in the channel, so I don't know if what you are implying would work. Regardless, I agree it doesn't make sense to join all the channels of every guild when you join it...

**Comment by tealkami** [comment]
Personally I don't want any group-chat/channels/"guilds" in Miranda at all , just private-chat/IM/DM syncing
Before you could turn off subgroups , it made even bigger mess in my Miranda

BTW you understand NUMBER 3000 (goup chat) contacts ? after 15years using Miranda IM/NG in all kind networks I don't have above 1000contacts , and I use like 3. version of Skype on Miranda 
sometimes it crash because of I go online on Discord (I do it once a month for sync)

**Comment by georgehazan** [comment]
ok, in the dev build I committed a patch that allows you to toggle Enable/Disable sync for any guild contact. By default all new guilds from your list have disabled sync (i.e. on a new profile), for all existing guilds sync is enabled to keep plugin's behavior consistent. Enjoy

**Comment by tealkami** [comment]
thank you
